pkg/scheduler/framework/plugins/dynamicresources/extendeddynamicresources.go

```diff
@@ -245,6 +290,8 @@ func createDeviceRequests(pod *v1.Pod, extendedResources map[v1.ResourceName]int
 	// supported separately.
 	containers := slices.Clone(pod.Spec.InitContainers)
 	containers = append(containers, pod.Spec.Containers...)
+
+	containerResourceIndices := buildContainerResourceIndices(containers)
```

Comment:
plesae note: 
in PR: https://github.com/kubernetes/kubernetes/pull/134882

we merged createDeviceRequests and createRequestMappings into one function, and that is called at the Filter phase, then the request mapping is stored in cycle state, and retrieved at PreBind phase, we have to do that to make it easy to account for init containers' requests.

Reply:
regardless the buildContainerResourceIndices will be usable in that PR. Instead of sorting through every time, we sort once and then generate the indices